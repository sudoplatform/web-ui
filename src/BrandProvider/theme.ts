import { DefaultTheme } from 'styled-components'

const defaultColors = {
  primary: '#3770f6',
  success: '#1ccd82',
  error: '#ff403d',
  interactive: '#3770f6',
}

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: typeof defaultColors
    grays: {
      gray5: string
      gray10: string
      gray20: string
      gray40: string
      gray90: string
    }
    spacing: {
      small: string
      medium: string
      large: string
    }
    padding: {
      small: string
      medium: string
      large: string
    }
    borderRadius: {
      small: string
      medium: string
      large: string
    }
    borderWidth: {
      light: string
      medium: string
      heavy: string
    }
    fontSize: {
      body: string
      h1: string
      h2: string
      h3: string
    }
  }
}

export type BrandColors = typeof defaultColors

export const defaultTheme: DefaultTheme = {
  colors: defaultColors,
  grays: {
    gray5: '#f5f5f7',
    gray10: '#dde1e6',
    gray20: '#c0c9d4',
    gray40: '#656d7f',
    gray90: '#0f1111',
  },
  spacing: {
    small: '6px',
    medium: '10px',
    large: '20px',
  },
  padding: {
    small: '6px',
    medium: '10px',
    large: '20px',
  },
  borderRadius: {
    small: '4px',
    medium: '8px',
    large: '12px',
  },
  borderWidth: {
    light: '0.5px',
    medium: '1px',
    heavy: '2px',
  },
  fontSize: {
    body: '12px',
    h1: '24px',
    h2: '18px',
    h3: '15px',
  },
}
