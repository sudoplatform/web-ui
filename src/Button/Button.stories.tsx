import { storiesOf } from '@storybook/react'
import React from 'react'

import { Button } from './Button'

storiesOf('Button', module)
  .add('kind=default', () => <Button kind="default">Default</Button>)
  .add('kind=primary', () => <Button kind="primary">Primary</Button>)
  .add('kind=default-danger', () => (
    <Button kind="default" danger={true}>
      Danger Default
    </Button>
  ))
  .add('kind=primary-danger', () => (
    <Button kind="primary" danger={true}>
      Danger Primary
    </Button>
  ))
  .add('kind=link', () => <Button kind="link">Link</Button>)
  .add('kind=link-danger', () => (
    <Button kind="link" danger={true}>
      Danger Link
    </Button>
  ))
  .add('clickable', () => (
    <Button onClick={() => alert('Clicked')}>Click Me</Button>
  ))
  .add('loading', () => <Button loading={true}>Loading</Button>)
  .add('disabled', () => <Button disabled={true}>Disabled</Button>)
