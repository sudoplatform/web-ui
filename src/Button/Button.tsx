import { Button as AntdButton } from 'antd'
import Color from 'color'
import React from 'react'
import styled from 'styled-components'

const StyledButton = styled(AntdButton)`
  &.ant-btn-primary {
    border-radius: ${({ theme }) => theme.borderRadius.small};
    border-width: ${({ theme }) => theme.borderWidth.light};
    background-color: ${({ theme }) => theme.colors.primary};
    border-color: ${({ theme }) => theme.colors.primary};
    color: #fff;

    :active,
    :hover,
    :focus {
      background-color: ${({ theme }) => theme.colors.primary};
      border-color: ${({ theme }) => theme.colors.primary};
      box-shadow: 0px 0px 5px 2px
        ${({ theme }) => Color(theme.colors.primary).alpha(0.4).rgb().string()};
      color: #fff;
    }
  }

  &.ant-btn-default {
    border-radius: ${({ theme }) => theme.borderRadius.small};
    border-width: ${({ theme }) => theme.borderWidth.light};
    background-color: #fff;
    border-color: ${({ theme }) => theme.colors.primary};
    color: ${({ theme }) => theme.colors.primary};

    :active,
    :hover,
    :focus {
      background-color: #fff;
      border-color: ${({ theme }) =>
        Color(theme.colors.primary).darken(0.5).hex()};
      box-shadow: 0px 0px 5px 2px
        ${({ theme }) => Color(theme.colors.primary).alpha(0.4).rgb().string()};
      color: ${({ theme }) => Color(theme.colors.primary).darken(0.25).hex()};
    }
  }

  &.ant-btn-link {
    color: ${({ theme }) => theme.colors.primary};

    :active,
    :not([disabled]):hover,
    :focus {
      color: ${({ theme }) => Color(theme.colors.primary).darken(0.25).hex()};

      > span {
        text-decoration: underline;
      }
    }
  }

  &.ant-btn-default.ant-btn-dangerous {
    background-color: #fff;
    border-color: ${({ theme }) => theme.colors.error};
    color: ${({ theme }) => theme.colors.error};

    :active,
    :hover,
    :focus {
      background-color: #fff;
      border-color: ${({ theme }) =>
        Color(theme.colors.error).darken(0.5).hex()};
      box-shadow: 0px 0px 5px 2px
        ${({ theme }) => Color(theme.colors.error).alpha(0.4).rgb().string()};
      color: ${({ theme }) => Color(theme.colors.error).darken(0.25).hex()};
    }
  }

  &.ant-btn-primary.ant-btn-dangerous {
    background-color: ${({ theme }) => theme.colors.error};
    border-color: ${({ theme }) => theme.colors.error};
    color: #fff;

    :active,
    :hover,
    :focus {
      background-color: ${({ theme }) => theme.colors.error};
      border-color: ${({ theme }) => theme.colors.error};
      box-shadow: 0px 0px 5px 2px
        ${({ theme }) => Color(theme.colors.error).alpha(0.4).rgb().string()};
      color: #fff;
    }
  }

  &.ant-btn-link.ant-btn-dangerous {
    color: ${({ theme }) => theme.colors.error};

    :active,
    :not([disabled]):hover,
    :focus {
      color: ${({ theme }) => Color(theme.colors.error).darken(0.25).hex()};

      > span {
        text-decoration: underline;
      }
    }
  }

  [ant-click-animating-without-extra-node]:after {
    animation: none !important;
  }

  &.ant-btn-loading-icon {
    margin-right: 8px;
  }
`

interface Props {
  className?: string
  kind?: 'primary' | 'default' | 'link' | 'ghost'
  danger?: boolean
  loading?: boolean
  disabled?: boolean
  type?: 'submit' | 'button' | 'reset'
  onClick?: React.MouseEventHandler<HTMLElement>
}

export const Button: React.FC<Props> = (props) => {
  const { kind, type, disabled, ...rest } = props

  return (
    <StyledButton
      htmlType={type}
      type={disabled ? 'ghost' : kind ?? 'default'}
      disabled={disabled}
      {...rest}
    />
  )
}
