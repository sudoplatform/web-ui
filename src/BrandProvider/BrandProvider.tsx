import 'antd/dist/antd.min.css'

import React from 'react'
import { ThemeProvider } from 'styled-components'

import { BrandColors, defaultTheme } from './theme'

interface Props {
  colors?: BrandColors
}

export const BrandProvider: React.FC<Props> = (props) => {
  const theme = {
    ...defaultTheme,
    colors: {
      ...defaultTheme.colors,
      ...props.colors,
    },
  }

  return <ThemeProvider theme={theme}>{props.children}</ThemeProvider>
}
