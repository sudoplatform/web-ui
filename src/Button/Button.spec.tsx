import React from 'react'
import { create } from 'react-test-renderer'

import { BrandProvider } from '../BrandProvider'
import { Button } from './Button'

describe('Button', () => {
  it('should render', () => {
    const wrapper = create(
      <BrandProvider>
        <Button>TEST</Button>
      </BrandProvider>,
    )
    const serialized = wrapper.toJSON()
    expect(serialized).toMatchSnapshot()
  })
})
