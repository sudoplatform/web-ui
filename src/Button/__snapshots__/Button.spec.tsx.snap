// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
.c0.ant-btn-primary {
  border-radius: 4px;
  border-width: 0.5px;
  background-color: chocolate;
  border-color: chocolate;
  color: #fff;
}

.c0.ant-btn-primary:active,
.c0.ant-btn-primary:hover,
.c0.ant-btn-primary:focus {
  background-color: chocolate;
  border-color: chocolate;
  box-shadow: 0px 0px 5px 2px rgba(210,105,30,0.4);
  color: #fff;
}

.c0.ant-btn-default {
  border-radius: 4px;
  border-width: 0.5px;
  background-color: #fff;
  border-color: chocolate;
  color: chocolate;
}

.c0.ant-btn-default:active,
.c0.ant-btn-default:hover,
.c0.ant-btn-default:focus {
  background-color: #fff;
  border-color: #69350F;
  box-shadow: 0px 0px 5px 2px rgba(210,105,30,0.4);
  color: #9D4F17;
}

.c0.ant-btn-link {
  color: chocolate;
  border: 0;
}

.c0.ant-btn-link:active,
.c0.ant-btn-link:not([disabled]):hover,
.c0.ant-btn-link:focus {
  color: #9D4F17;
}

.c0.ant-btn-link:active > span,
.c0.ant-btn-link:not([disabled]):hover > span,
.c0.ant-btn-link:focus > span {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0.ant-btn-default.ant-btn-dangerous {
  background-color: #fff;
  border-color: fuchsia;
  color: fuchsia;
}

.c0.ant-btn-default.ant-btn-dangerous:active,
.c0.ant-btn-default.ant-btn-dangerous:hover,
.c0.ant-btn-default.ant-btn-dangerous:focus {
  background-color: #fff;
  border-color: #80007F;
  box-shadow: 0px 0px 5px 2px rgba(255,0,255,0.4);
  color: #BF00BF;
}

.c0.ant-btn-primary.ant-btn-dangerous {
  background-color: fuchsia;
  border-color: fuchsia;
  color: #fff;
}

.c0.ant-btn-primary.ant-btn-dangerous:active,
.c0.ant-btn-primary.ant-btn-dangerous:hover,
.c0.ant-btn-primary.ant-btn-dangerous:focus {
  background-color: fuchsia;
  border-color: fuchsia;
  box-shadow: 0px 0px 5px 2px rgba(255,0,255,0.4);
  color: #fff;
}

.c0.ant-btn-link.ant-btn-dangerous {
  color: fuchsia;
}

.c0.ant-btn-link.ant-btn-dangerous:active,
.c0.ant-btn-link.ant-btn-dangerous:not([disabled]):hover,
.c0.ant-btn-link.ant-btn-dangerous:focus {
  color: #BF00BF;
}

.c0.ant-btn-link.ant-btn-dangerous:active > span,
.c0.ant-btn-link.ant-btn-dangerous:not([disabled]):hover > span,
.c0.ant-btn-link.ant-btn-dangerous:focus > span {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 [ant-click-animating-without-extra-node]:after {
  -webkit-animation: none !important;
  animation: none !important;
}

.c0.ant-btn-loading-icon {
  margin-right: 8px;
}

<button
  className="ant-btn ant-btn-default c0"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <span>
    TEST
  </span>
</button>
`;
